### PR TITLE
Allow to open a filedes and FILE* as SQFile*

### PR DESCRIPTION
**Pull request by zecke**
When using BSD/Linux/Posix APIs a filedescriptor will be returned from
the system. Make it possible to turn either a filedes or a FILE\* into a
SQFile*. This can be used to read/write using the StandardFileStream and
its subclasses.

A simple test to open filedes '2' and write to it has worked. More
advanced modes are still missing.

### CODE REVIEW

File: platforms/Cross/plugins/FilePlugin/sqFilePluginBasicPrims.c [hunk 1]
```diff
@@ -302,10 +303,43 @@ sqFileFdOpen(SQFile *sqFile, int fd, sqInt writeFlag)
 	return sqFileFileOpen(sqFile, file, writeFlag);
 }
 
+static int
+isFile(FILE *file)
```
Comment (zecke) [inline platforms/Cross/plugins/FilePlugin/sqFilePluginBasicPrims.c]: Good idea but with the degenerated fread() for line buffering let's not merge this patch!

**Comment by takano32** [comment]
@zecke Would you merge this PR for auto-merge?
https://github.com/zecke/pharo-vm/pull/1

**Comment by zecke** [comment]
What do you mean with automerge? I want to try similar changes into OpenSmalltalk first but there is some reluctance in adding anything to the FilePlugin...

**Comment by takano32** [comment]
@zecke Sorry, not auto-merge but pass-CI.
I've used 'auto-merge' as meaning for merging in GitHub Web interface.

This PR's CI Fails are now gone with merging trunk.
And merging trunk PR is https://github.com/zecke/pharo-vm/pull/1 .

screen shot 2016-09-30 at 19 06 48

**Comment by takano32** [comment]
Thanks.
Now, this PR have passed CI.
And LGTM.

**Comment by estebanlm** [comment]
I will include this in the new master... once I can manage to have one :)

**Comment by marianopeck** [comment]
Cool. Thanks Esteban. Could you please let us know here when a VM is finally released including this PR? Thanks!

**Comment by marianopeck** [comment]
I think this PR was never integrated into the VM or it might have been lost. Can you confirm this please?

**Comment by zecke** [comment]
It is open. The Squeak-VM devs were reluctant to add it but now we can habve cppIf... to have this pharo specific.

**Comment by estebanlm** [comment]
yes, this is the problem. 
we need a "pharo-only" solution... now, some questions: 

1) can't this be done using UFFI?
2) if not... can you make it work just for Pharo? (then I will merge it)

**Comment by zecke** [comment]
1. It can't be done _right_ now and before doing major changes to the VM IO handling there should be some buy-in.

2. For Pharo7 let's have the primitive. We can improve FileStream and Socket later.

**Comment by zecke** [comment]
@estebanlm please re-open it.

**Comment by estebanlm** [comment]
I can't, this is against a branch that no longer exists.

**Comment by marianopeck** [comment]
Looks like this PR was moved to this one: https://github.com/pharo-project/pharo-vm/pull/142

**Pull request closed without merging**